api/src/search/types.ts

```diff
@@ -3,8 +3,18 @@ import { ContributorEntity } from "@dzcode.io/models/dist/contributor";
 import { GeneralResponse } from "src/app/types";
 import { MultiSearchResponse } from "meilisearch";
 import { ProjectEntity } from "@dzcode.io/models/dist/project";
+import { IsNotEmpty, IsPositive, IsString } from "class-validator";
 
-export interface GetSearchResponse extends GeneralResponse {
+export class SearchQuery {
+  @IsString()
+  @IsNotEmpty()
+  query!: string;
+
+  @IsPositive()
+  limit?: number = 5;
```

Comment:
so that the types are correct:

```suggestion
  limit = 5;
```

Reply:
:')